pkg/serviceaccount/claims.go

```diff
@@ -166,6 +183,17 @@ func (v *validator) Validate(_ string, public *jwt.Claims, privateObj interface{
 		podUID = podref.UID
 	}
 
+	// Check special 'warnafter' field for projected service account token transition.
+	warnafter := private.Kubernetes.WarnAfter
+	if warnafter != 0 {
+		if now.After(warnafter.Time()) {
+			klog.Warningf("Stale service account token used, service account: %s", public.Subject)
+			metrics.StaleTokensTotal.Inc()
+		} else {
+			metrics.ValidTokensTotal.Inc()
```

Comment:
is this a useful metric? even in the presence of poorly behaved clients that do not reload tokens, they will trigger this metric during the first hour they are running

Reply:
It would be used as a reference for cluster admin. After fixing the client, if they observe valid token counter grows as usual but stale token counter does not grow, then everything is fine.
So it's generally just gives them some confidence, and yes the metric would grow at the first hour of bad tokens.